Corona-Warn-App/src/main/java/de/rki/coronawarnapp/task/TaskState.kt

```diff
@@ -0,0 +1,47 @@
+package de.rki.coronawarnapp.task
+
+import org.joda.time.Instant
+import kotlin.reflect.KClass
+
+interface TaskState {
+    val request: TaskRequest
+    val createdAt: Instant
+    val startedAt: Instant?
+    val completedAt: Instant?
+    val error: Throwable?
+    val result: Task.Result?
+    val state: State
```

Comment:
state within a state might be confusing. Maybe  "progress" or "executionState" to be moe precise?

Reply:
Will change it to `executionState`.